src/meltano/core/tracking/tracker.py

```diff
@@ -0,0 +1,196 @@
+        """
+        try:
+            return tzlocal.get_localzone_name()
+        except Exception:
+            return datetime.datetime.now().astimezone().tzname()
+
+    @contextmanager
+    def with_contexts(self, *extra_contexts) -> Tracker:
+        """Context manager within which the `Tracker` has additional Snowplow contexts.
+
+        Args:
+            extra_contexts: The additional contexts to add to the `Tracker`.
+
+        Yields:
+            A `Tracker` with additional Snowplow contexts.
+        """
+        prev_contexts = self.contexts
+        self.contexts = (*prev_contexts, *extra_contexts)
+        try:
+            yield self
+        finally:
+            self.contexts = prev_contexts
+
+    def can_track(self) -> bool:
+        """Check if the tracker can be used.
+
+        Returns:
+            True if the tracker can be used, False otherwise.
+        """
+        return self.snowplow_tracker is not None and self.send_anonymous_usage_stats
+
+    def track_struct_event(self, category: str, action: str) -> None:
+        """Fire a structured tracking event.
+
+        Note: This is a legacy method that will be removed in a future version, once LegacyTracker is no longer used.
+
+        Args:
+            category: The category of the event.
+            action: The event actions.
+        """
+        if not self.can_track():
+            return
+
+        if self.project.active_environment is not None:
+            action = f"{action} --environment={hash_sha256(self.project.active_environment.name)}"
+
+        try:
+            self.snowplow_tracker.track_struct_event(
+                category=category,
+                action=action,
+                label=self.project_id,
+            )
+        except Exception as ex:
+            logger.warning(
+                f"Failed to submit struct event to Snowplow:\n{traceback.format_exception(ex)}"
+            )
+
+    def track_unstruct_event(self, event_json: SelfDescribingJson) -> None:
+        """Fire an unstructured tracking event.
+
+        Args:
+            event_json: The SelfDescribingJson event to track. See the Snowplow documentation for more information.
+        """
+        if not self.can_track():
+            return
+        try:
+            self.snowplow_tracker.track_unstruct_event(event_json, self.contexts)
+        except Exception as ex:
+            logger.warning(
+                f"Failed to submit unstruct event to Snowplow:\n{traceback.format_exception(ex)}"
+            )
+
+    def track_command_event(self, event_json: dict[str, Any]) -> None:
+        """Fire generic command tracking event.
+
+        Args:
+            event_json: The event JSON to track. See cli_event schema for more details.
+        """
+        self.track_unstruct_event(
+            SelfDescribingJson(
+                "iglu:com.meltano/cli_event/jsonschema/1-0-0", event_json
```

Comment:
Here, and everywhere we have something like `"iglu:.../1-0-0"`, what do you think of moving these to constant string declarations at the top of the page.

As a single constant it might be something like `COMMAND_EVENT_SCHEMA_IGLU`.

Or as a two-part, it might be something like `COMMAND_EVENT_SCHEMA_IGLU` and `COMMAND_EVENT_SCHEMA_IGLU_VERSION`

Reply:
A constant dictionary mapping schema names to their version might make sense for this.